### PR TITLE
Replace by fee functionality

### PR DESCRIPTION
**Pull request by aulneau**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/blockstack/ux/actions/runs/1187452437).

This PR works on top of https://github.com/blockstack/stacks-wallet-web/pull/1639 to implement the actual replace-by-fee functionality. #1639 made it possible to modify the fee/nonce of a given transaction. This PR makes it so a user can select a pending transaction and submit a duplicate transaction with only the fee changed.

https://user-images.githubusercontent.com/11803153/131561394-c8dbb46c-8dc0-4ef9-8ff0-36640508df06.mp4

cc/ @aulneau @kyranjamie @fbwoolf

### CODE REVIEW

File: src/components/popup/tx-item.tsx [hunk 1]
```diff
@@ -88,8 +90,41 @@ const Status: React.FC<{ transaction: Tx } & BoxProps> = ({ transaction, ...rest
...
+const SpeedUpButton = ({
+  txid,
+  isHovered,
+  isEnabled,
+}: {
+  txid: string;
+  isHovered: boolean;
+  isEnabled: boolean;
+}) => {
+  const [rawTxId, setTxId] = useRawTxIdState();
```
Comment (fbwoolf) [inline src/components/popup/tx-item.tsx:102]: I think later you use `setRawTxId` so maybe keep same here?

File: src/features/speed-up-transaction-drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+const FeeForm = () => {
+  const refreshAccountData = useRefreshAllAccountData();
+  const [multiplier] = useFeeRateMultiplier();
+  const [multiplierCustom] = useFeeRateMultiplierCustom();
+  const [, setUseCustom] = useFeeRateUseCustom();
+  const [, setFeeRate] = useFeeRate();
+  const tx = useSelectedTx();
+  const rawTx = useRawStacksTransactionState();
+  const byteSize = useRawStacksTransactionByteSizeState();
+  const [, setTxId] = useRawTxIdState();
```
Comment (fbwoolf) [inline src/features/speed-up-transaction-drawer.tsx:134]: Same here.

File: src/components/popup/tx-item.tsx (hunk 1 above)
Comment (aulneau) [reply src/components/popup/tx-item.tsx:102]: will make changes in follow commit, going to merge!

**Review by kyranjamie — APPROVED** [review]
Great stuff Thomas, looks like it works really well. 

I'd go for some more descriptive component/method names, but no biggie for now. 

Can test/review in more depth tomorrow morning. Let me know if there's anything inparticular for me and Beguene to pick up in the morning.

**Comment by kyranjamie** [comment]
lol, post merge review

**Pull request merged**